cmd/branchingconfigmanagers/tide-config-manager/main.go

```diff
@@ -237,6 +237,136 @@ func (cfe codeFreezeEvent) GetDataFromProwConfig(pc *prowconfig.ProwConfig) {
+	releaseFuture                 string
+	openshiftFuture               string
+	past                          string
+	current                       string
+	future                        string
+}
+
+func newGeneralAvailabilityEvent(past, current, future string) generalAvailabilityEvent {
+	noXYAllowList := sets.NewString(
+		// Logging team
+		"openshift/cluster-logging-operator",
+		"openshift/elasticsearch-operator",
+		"openshift/elasticsearch-proxy",
+		"openshift/origin-aggregated-logging",
+		// Does not seem to be even branched, likely does not need this config
+		"openshift/app-netutil",
+		"openshift/network-tools",
+		"openshift-priv/app-netutil",
+		"openshift-priv/network-tools",
+	)
+	excludedAllowList := sets.NewString(
+		// Assisted
+		"openshift/assisted-installer",
+		"openshift/assisted-installer-agent",
+		"openshift/assisted-test-infra",
+		"openshift/assisted-image-service",
+		"openshift/assisted-service",
+		// Weird but consistent, no OCP criteria
+		"openshift/windows-machine-config-bootstrapper",
+		"openshift/windows-machine-config-operator",
+		"openshift-priv/windows-machine-config-bootstrapper",
+		"openshift-priv/windows-machine-config-operator",
+		"red-hat-storage/ceph-csi",
+	).Union(noXYAllowList)
+
+	return generalAvailabilityEvent{
+		repos:                         sets.NewString(),
+		excludedAllowList:             excludedAllowList,
+		noXYAllowList:                 noXYAllowList,
+		openshiftReleaseBranches:      sets.NewString(release+current, openshift+current),
+		openshiftReleaseBranchesPlus1: sets.NewString(release+future, openshift+future),
+		releasePast:                   release + past,
+		releaseCurrent:                release + current,
+		releaseFuture:                 release + future,
+		openshiftPast:                 openshift + past,
+		openshiftCurrent:              openshift + current,
+		openshiftFuture:               openshift + future,
+		past:                          past,
+		current:                       current,
+		future:                        future,
+	}
+}
+
+func (gae generalAvailabilityEvent) ModifyQuery(q *prowconfig.TideQuery, repo string) {
+	gae.ensureStaffEngApprovedLabel(q)
+	gae.ensureCherryPickApprovedLabel(q)
+	gae.overrideExcludedBranches(q)
+}
+
+func (gae generalAvailabilityEvent) GetDataFromProwConfig(*prowconfig.ProwConfig) {}
+
+func (gae *generalAvailabilityEvent) ensureCherryPickApprovedLabel(q *prowconfig.TideQuery) {
+	reqLabels := sets.NewString(q.Labels...)
+	branches := sets.NewString(q.IncludedBranches...)
+	if reqLabels.Has(cherryPickApproved) {
+		if branches.Has(gae.releasePast) {
+			branches.Insert(gae.releaseCurrent)
+		}
+		if branches.Has(gae.openshiftPast) {
+			branches.Insert(gae.openshiftCurrent)
+		}
+		if branches.Intersection(gae.openshiftReleaseBranches).Len() == 0 && !gae.noXYAllowList.Has(q.Repos[0]) {
+			fmt.Printf("Suspicious cherry-pick-approved query (without %s): %s\n", gae.current, q.Repos)
```

Comment:
I will create a follow-up card to get rid of these prints that I have here, but they are complicated cases that need further analysis. Usually, it means the config is broken, or something is missing. This requires closer investigation.